src/main/resources/resources/cohortdefinition/sql/generateCohort.sql

```diff
@@ -56,13 +56,11 @@ from #included_events;
 
 @censoringInserts
 
-
-
 with collapse_constructor_input (person_id, start_date, end_date) as
 (
 	select F.person_id, F.start_date, F.end_date
 	FROM (
-	  select I.event_id, I.person_id, I.start_date, E.end_date, dense_rank() over (partition by I.person_id, I.event_id order by E.end_date) as ordinal 
+	  select I.event_id, I.person_id, I.start_date, E.end_date, row_number() over (partition by I.person_id, I.event_id order by E.end_date) as ordinal 
```

Comment:
@gowthamrao : this is one change that wasn't to revert dense_rank(): I think in this case row_number() is correct: the query is saying 'give me the first end date that happens on or after the qualified event'.   dense_rank() would allow possibly 2 records of the same end_date have ordinal = 1, while row_number() makes sure there's just 1.  dense_rank() vs. row_number() here actaully won't change the era collapsing logic (it will just have a second record to collapse into the same era) but I wanted to make sure I wasn't mising something else.  I think in cases where you want to say 'select exactly one record' you want to use row_number() = 1 where if you want to say 'select all rows that share the earliest date' you would use dense_rank() = 1.